src/utilities/TypeInfo.ts

```diff
@@ -178,15 +203,50 @@ export class TypeInfo {
       case Kind.ARGUMENT: {
         let argDef;
         let argType: unknown;
-        const fieldOrDirective = this.getDirective() ?? this.getFieldDef();
-        if (fieldOrDirective) {
-          argDef = fieldOrDirective.args.find(
-            (arg) => arg.name === node.name.value,
+        const directive = this.getDirective();
+        const fragmentSpread = this._fragmentSpread;
+        const fieldDef = this.getFieldDef();
+        if (directive) {
+          argDef = directive.args.find((arg) => arg.name === node.name.value);
+        } else if (fragmentSpread) {
+          const fragmentDef =
+            this._fragmentDefinitions.get(fragmentSpread.name.value);
+          const fragVarDef = fragmentDef?.variableDefinitions?.find(
+            (varDef) => varDef.variable.name.value === node.name.value,
           );
-          if (argDef) {
-            argType = argDef.type;
+          if (fragVarDef) {
+            const fragVarType = typeFromAST(schema, fragVarDef.type);
+            if (isInputType(fragVarType)) {
+              const fragVarDefault = fragVarDef.defaultValue
+                ? valueFromAST(fragVarDef.defaultValue, fragVarType)
+                : undefined;
+
+              // Minor hack: transform the FragmentArgDef
```

Comment:
Existing discussion https://github.com/graphql/graphql-js/pull/3835/files#r1101008869

Reply:
We've addressed the comments in that prior discussion by introducing the new GraphQLVariableSignature type corresponding to GraphQLArgument, i.e. a structure that defines an argument definition, where the former is for variables, and the latter is for arguments defined by the schema/AST, i.e. fields and directives. A union of those two types can now be taken by the `experimentalGetArgumentValues()` function. We have also introduced new helper functions on TypeInfo and ValidationContext to distinguish between the types of arguments.